locator/tablets.cc

```diff
@@ -140,6 +144,25 @@ tablet_migration_streaming_info get_migration_streaming_info(const locator::topo
             });
 
             return result;
+        case tablet_transition_kind::rebuild_v2: {
+            if (!trinfo.pending_replica.has_value()) {
+                return result; // No nodes to stream to -> no nodes to stream from
+            }
+
+            auto s = std::unordered_set<tablet_replica>(trinfo.next.begin(), trinfo.next.end());
+            erase_if(s, [&] (const tablet_replica& r) {
+                if (r == *trinfo.pending_replica) {
```

Comment:
Why do we  return false for the pending replica node explicitly?  The pending node should not be excluded nor absent in the topo.

Reply:
Replicas participating in the rebuild repair are chosen based on `migration_streaming_info::written_to`. We cannot exclude pending replica from that set - if it's dead then repair has to fail.

Moreover, `check_excluded_replicas()` will capture when pending replica is dead and rollback the rebuild

Reply:
Yes, I know we want the pending_replica to be included in s. My argument is that even if we do not return false in the filter, pending_replica will not be removed from the list, because the pending_replica node is not supposed to be marked as excluded when the rebuild transition runs. 

To ensure the pending_replica in the list, I think it is better to add it explicitly after the filter.